vector/src/test/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewModelTest.kt

```diff
@@ -477,10 +484,28 @@ class SessionOverviewViewModelTest {
                 aPusher(deviceId = "another id", enabled = false)
         )
         fakeSession.pushersService().givenPushersLive(pushers)
+        fakeSession.pushersService().givenGetPushers(pushers)
 
         val viewModel = createViewModel()
         viewModel.handle(SessionOverviewAction.TogglePushNotifications(A_SESSION_ID_1, true))
 
-        fakeSession.pushersService().verifyOnlyTogglePusherCalled(pushers.first(), true)
+        fakeSession.pushersService().verifyTogglePusherCalled(pushers.first(), true)
+    }
+
+    @Test
+    fun `when handle TogglePushNotifications, then update account local notification settings`() {
+        val pushers = listOf(
+                aPusher(deviceId = A_SESSION_ID_1, enabled = false),
+                aPusher(deviceId = "another id", enabled = false)
+        )
+        fakeSession.pushersService().givenPushersLive(pushers)
+
+        val viewModel = createViewModel()
+        viewModel.handle(SessionOverviewAction.TogglePushNotifications(A_SESSION_ID_1, true))
+
+        fakeSession.accountDataService().verifyUpdateUserAccountDataEventSucceeds(
+                UserAccountDataTypes.TYPE_LOCAL_NOTIFICATION_SETTINGS + A_SESSION_ID_1,
+                mapOf("is_silenced" to false),
```

Comment:
Could you also use the `LocalNotificationSettingsContent` class instead of a map?